src/chunk_index.c

```diff
@@ -863,8 +891,11 @@ chunk_index_tuple_rename(TupleInfo *ti, void *data)
 	else
 		namestrcpy(&chunk_index->index_name, info->newname);
 
-	ts_catalog_update(ti->scanrel, tuple);
-	heap_freetuple(tuple);
+	ts_catalog_update(ti->scanrel, new_tuple);
+	heap_freetuple(new_tuple);
+
+	if (should_free)
+		heap_freetuple(tuple);
```

Comment:
Can it be freed directly after copying to `new_tuple` or use `materialise = true` instead of `new_tuple`?